spec/classes/init_spec.rb

```diff
@@ -570,6 +570,16 @@
           end
           it { should contain_file('rsyslog_config').with_content(/^\$DirCreateMode 0770$/) }
         end
+
+	context 'with msg_reduction=true' do
```

Comment:
Good test, though we also need if msg_reduction is false. Since we allow stringified booleans, `'true'` and `'false'`, we should check those as well. Try this approach.

```ruby
describe 'with msg_reduction' do
  [true,'true'].each do |value|
    context "set to #{value.class} #{value}" do
      let (:params) { { :msg_reduction => value } }
      ... put tests here ...
    end
  [false,'false'].each do |value|
    context "set to #{value.class} #{value}" do
      let (:params) { { :msg_reduction => value } }
      ... put tests here ...
    end
  end
end
```